sqlagg/columns.py

```diff
@@ -88,15 +85,37 @@ def sql_column(self):
 
 class SumWhen(ConditionalAggregation):
     """
-    SumWhen("vehicle", whens={"unicycle": 1, "bicycle": 2, "car": 4}, else_=0, alias="num_wheels")
+    SumWhen("vehicle", whens=[["unicycle", 1], ["bicycle", 2], ["car", 4]], else_=0, alias="num_wheels")
     """
     aggregate_fn = func.sum
 
 
+class SumWhenWithBinds(SumWhen):
+    """
+    SumWhen("age_cohort",
+            whens=[
+                ["when": "age < :ten", "then": 1],
+                ["when": "age < :thirty", "then": 2],
+            binds={
+                "ten": 10,
+                "thirty": 30,
+            },
+            else_=0)
+    """
+    def __init__(self, key=None, whens=None, binds=None, else_=None, *args, **kwargs):
+        super(SumWhenWithBinds, self).__init__(key, whens, else_, *args, **kwargs)
+        self.binds = binds or {}
+
+    @property
+    def sql_column(self):
+        return ConditionalTemplateColumn(self.key, self.whens, self.binds, self.else_,
+                                         self.aggregate_fn, self.alias)
+
+
 class ConditionalColumn(SqlColumn):
     """
     ConditionalColumn("vehicle",
-                      whens={"unicycle": 1, "bicycle": 2, "car": 4},
+                      whens=[["unicycle", 1], ["bicycle": 2], ["car": 4]],
```

Comment:
I'm not sure if this should be a list of lists (like the example in docstring of `SumWhen`) or a list of dictionaries